ui/widgets/nuxeo-collapsible.js

```diff
@@ -0,0 +1,134 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import '@polymer/iron-collapse/iron-collapse.js';
+import { mixinBehaviors } from '@polymer/polymer/lib/legacy/class.js';
+import '@polymer/iron-flex-layout/iron-flex-layout.js';
+import '@polymer/iron-icon/iron-icon.js';
+import '@polymer/iron-icons/hardware-icons.js';
+import '@nuxeo/nuxeo-elements/nuxeo-element.js';
+import '@polymer/polymer/lib/elements/dom-if.js';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+
+{
+  /**
+   * An element for creating a collapsible block of content.
+   * It comes with a default chevron that can be displayed at the end (default) or at the beggining of the heading.
+   * The heading can be expanded by default.
+   * The chevron has its own slot so it can be also customized.
+   * It is also possible to disable the default animation.
+   *
+   * Example:
+   *
+   *     <nuxeo-collapsible chevron chevronAtStart opened>
+   *        <label slot="heading">Nuxeo</label>
+   *        <div>Collapsed content</div>
+   *     </nuxeo-collapsible>
+   *
+   * ### Styling
+   * The following custom properties and mixins are available for styling:
+   * Custom property | Description | Default
+   * ----------------|-------------|----------
+   * `--nuxeo-collapsible-heading-padding` | Heading padding | `4px`
+   * `--nuxeo-collapsible-icon-height` | Chevron height | `20px`
+   * `--nuxeo-collapsible-icon-width` | Chevron width | `20px`
+   *
+   * @memberof Nuxeo
+   *
+   */
+
+  class Collapsible extends mixinBehaviors([], Nuxeo.Element) {
+    static get template() {
+      return html`
+        <style>
+          button {
+            height: 100%;
+            width: 100%;
+            display: flex;
+            justify-content: space-between;
+            align-items: center;
+            margin: 0;
+            padding: var(--nuxeo-collapsible-heading-padding, 4px);
+            background: transparent;
+            border: none;
+            cursor: pointer;
+            text-decoration: none;
+            font-size: 1rem;
+          }
+          /* to fix a blinking default style on safari */
+          button:active {
+            color: inherit;
+          }
+          .heading {
+            width: calc(100% - var(--nuxeo-collapsible-icon-width, 20px));
+            overflow-x: hidden;
+            text-align: start;
+          }
+          iron-icon {
+            --iron-icon-height: var(--nuxeo-collapsible-icon-height, 20px);
+            --iron-icon-width: var(--nuxeo-collapsible-icon-width, 20px);
+          }
+          iron-collapse {
+            padding: 0 var(--nuxeo-collapsible-heading-padding, 4px);
+          }
+        </style>
+
+        <button aria-expanded="[[opened]]" on-tap="_toggle">
+          <slot class="heading" name="heading"></slot>
+          <iron-icon icon="[[_toggleIcon(opened)]]"></iron-icon>
```

Comment:
@Gabez0r I guess this is the crux of the matter, should we have a hardcoded icon ? At some point we have a property to move it to the beginning, at some point we may want to use a custom icon, etc... in the end it's not clear whether we should just have a `header` slot and let people do whatever they want (the tap is in the whole thing anyway...)

Reply:
So we need to clarify what uses cases might require the header to be overridden. I can imagine a few cases:
1) we want to fully customize the header (could be replaced with parts, mixins or a bunch of variables)
2) people might want to move the chevron to the right (that's what the html details/summary combo do)
3) people what to add some sort of actions there

I just don't know how common any of these would be. Probably the first one would be the most common? If so, we could expose a `part` (or mixin) for that and probably add a [template extension point](https://polymer-library.polymer-project.org/3.0/docs/devguide/dom-template#insertcontent) for the whole header, allowing cases 2 and 3 to be done through code?